src/xml.c

```diff
@@ -1100,38 +1104,77 @@ lyxml_value_compare(const struct ly_ctx *ctx, const char *value1, void *val_pref
         return LY_ENOT;
     }
 
-    ptr1 = value1;
-    ptr2 = value2;
-    while (ptr1[0] && ptr2[0]) {
-        if (ptr1[0] != ptr2[0]) {
-            /* it can be a start of prefix that maps to the same module */
-            mod1 = mod2 = NULL;
-            if (val_prefix_data1 && (end1 = strchr(ptr1, ':'))) {
-                /* find module of the first prefix, if any */
-                mod1 = ly_resolve_prefix(ctx, ptr1, end1 - ptr1, LY_PREF_XML, val_prefix_data1);
+    if (!ctx2) {
+        ctx2 = ctx1;
+    }
+
+    ret = LY_SUCCESS;
+    for (value1_iter = value1, value2_iter = value2;
+            value1_iter && value2_iter;
+            value1_iter = value1_next, value2_iter = value2_next) {
+        value1_len = ly_string_next(value1_iter, NULL, &is_prefix1, &value1_next);
+        value2_len = ly_string_next(value2_iter, NULL, &is_prefix2, &value2_next);
+
+        if (is_prefix1 != is_prefix2) {
+            ret = LY_ENOT;
+            break;
+        }
+
+        if (is_prefix1 != 1) {
```

Comment:
Connected to a previous comment, it is a boolean value (even based on the name), so treat it that way.

Reply:
ok